django/db/models/fields/related_objects.py

```diff
@@ -0,0 +1,315 @@
+"""
+"Rel objects" for related fields.
+
+"Rel objects" (for lack of a better name) carry information about the relation
+modeled by a related field and provide some utility functions. They're stored
+in the ``rel`` attribute of the field.
```

Comment:
Oh wow I completely missed that commit of yours. Was it a first step to then turn the individual `Rel` objects into actual fields?